distributed/tests/test_client.py

```diff
@@ -1116,6 +1117,27 @@ def g():
     result = yield y._result()
     assert result == 456
 
+@gen_cluster(client=True)
+def test_upload_file_zip(c, s, a, b):
+    def g():
+        import myfile
+        return myfile.f()
+
+    with tmp_text('myfile.py', 'def f():\n    return 123') as fn_my_file, \
+         tmp_text('init.py', '') as fn_init:
+        try:
+            with zipfile.ZipFile('myfile.zip', 'w') as z:
+                for fn in (fn_my_file, fn_init):
+                    z.write(fn, arcname=os.path.basename(fn))
+            yield c._upload_file('myfile.zip')
+        finally:
+            if os.path.exists('myfile.zip'):
+                os.remove('myfile.zip')
+
+    sleep(1)  # TODO:  why is this necessary?
```

Comment:
There's no case where we call invalidate_cache in the upload_file method. 
Should I add it in all cases ? 
In the same way the sleep call is in all the upload_file tests.
Should I remove it from all the tests ?

```
                if ext in ('.py', '.pyc'):
                    logger.info("Reload module %s from .py file", name)
                    name = name.split('-')[0]
                    reload(import_module(name))
                if ext == '.egg':
                    import pkg_resources
                    sys.path.append(out_filename)
                    pkgs = pkg_resources.find_distributions(out_filename)
                    for pkg in pkgs:
                        logger.info("Load module %s from egg", pkg.project_name)
                        reload(import_module(pkg.project_name))
                    if not pkgs:
                        logger.warning("Found no packages in egg file")
                if ext == '.zip':
                    logger.info("Reload module %s from .zip file", name)
                    if out_filename not in sys.path:
                        sys.path.insert(0, out_filename)
                    invalidate_caches()
                    reload(import_module(name))
```